### ISSUE TITLE
Stop at start up screen "Preparing your library"

### ISSUE THREAD

**Issue by bao3** [tallbl0nde/TriPlayer#18]
I start TriPlayer, but it only show "  Preparing your library" . My SD card is 512G, and musics are in folder /music .

This screen keep for about 1hr ... What could I do ?

**Comment by tallbl0nde** [comment]
This bug is caused due to the file names of your songs. Check that they don't have "special characters" such as é or symbols from other languages. Once all these characters are removed it should load normally!

**Comment by bao3** [comment]
@tallbl0nde  Conformed that ... My songs' name contain chinese characters .

So could you patch it  ?

**Comment by tallbl0nde** [comment]
Hopefully, I'm still looking into it. Is there any chance you used sys-tune and these songs worked fine?

**Comment by bao3** [comment]
Tried , and sys-tune has the similar issue. The sys-tune could only read and load the song with Latin characters , and wouldnt list Chinese characters songs.

**Comment by tallbl0nde** [comment]
Thanks for testing that, I'm starting to think it's an issue with Horizon (the Switch's OS), so I'm not entirely sure it can be fixed but I'll see!

**Comment by Smoker1** [comment]
When you say Special Characters - in the actual File NAME ( "song Title - here.mp3" ) , or in the Description Details? I know with Nero WaveEdit, when you change something in a Audio File, it changes the Album description to that weird y Symbol, and another one, like it is in Chinese.
Yeah, also getting that Stuck at Building Library.

**Comment by tallbl0nde** [comment]
> When you say Special Characters - in the actual File NAME

Yes. The metadata within the song (album name, artist name, etc) can contain these characters, however the file name essentially has to only contain A-Z, 0-9, etc.

**Comment by Smoker1** [comment]
So no - in it, correct?

**Comment by Smoker1** [comment]
Also, I created a NSP if interested. That way it should be running on it's own instead of Homebrew Menu NSP
https://mega.nz/file/7J9EATKA#yqksbOSs3nZmeIjAawiAvNzlF6Bbiuia1BmI9gVSxGc

**Comment by tallbl0nde** [comment]
> So no - in it, correct?

A dash and any brackets should be fine, essentially if the symbol appears on [this table](http://www.asciitable.com/) (only the first one) it _should_ work.

**Comment by Smoker1** [comment]
So - should be ok, and [ ] are ok? So no ( ) in it? 
Example - 
Song Artist - Song Title (Instrumental).mp3

If that is OK, then that means I wont have that much to Edit. But if I can work with [ ], then it will be more to Edit, but not as much. LOL

**Comment by tallbl0nde** [comment]
I used all those characters in my ~1300 song library without issues, so they should be fine! :)

For example, the following name would work:
Artist - A Song Title (feat. Another Artist) [Remix].mp3

**Comment by Smoker1** [comment]
Might need a list of Symbols that will NOT work. I went through everything, and Edited the Filenames as much as I could look through. Still stuck. So far, based on your Example, these are allowed....
( ) [ ] . and - . 
any problems with the following?
, & _ ' " "

**Comment by tallbl0nde** [comment]
Unfortunately I can't give a definitive list as I don't know for sure which ones cause issues :P
However I'd say `&`, `,` and `_` would be fine, not sure about quotes though.

There's one way I found which files were causing issues for me at least:
1. Open a file explorer homebrew (I'll use N-Xplorer)
2. Navigate to `sdmc:/` and then `/music`
3. Look over all of your `.mp3`'s, if the file size reported seems wack, there's likely a bad character in the name

For example, when I replaced a character with `é` it said the size of my song was 2 bytes, which obviously isn't right.
